File: sgf-poker/sgf-poker-ios/Core/UseCases/FetchGamesUseCase.swift
// Core/UseCases/FetchGamesUseCase.swift

/// Retrieves the full ordered game list from the repository.
final class FetchGamesUseCase {

    private let repository: GameRepository

    init(repository: GameRepository) {
        self.repository = repository
    }

    /// - Returns: Games sorted by creation date, newest first.
    /// - Throws: `GameError.storageFailure` if the repository cannot be read.
    func execute() throws -> [Game] {
        try repository.fetchAll()
    }
}

File: sgf-poker-android/app/src/main/java/com/sgf/poker/usecases/game/FetchGamesUseCase.java
package com.sgf.poker.usecases.game;

import com.sgf.poker.data.repository.GameRepository;
import com.sgf.poker.domain.model.Game;
import java.util.List;

/** Retrieves all games sorted by event date descending. */
public class FetchGamesUseCase {

    private final GameRepository repository;

    public FetchGamesUseCase(GameRepository repository) {
        this.repository = repository;
    }

    public List<Game> execute() {
        return repository.fetchAll();
    }
}
